x-pack/plugins/gis/public/elasticsearch_geo_utils.test.js

```diff
@@ -260,96 +253,79 @@ describe('createExtentFilter', () => {
   it('should convert bounds to envelope', () => {
     const bounds = {
-      maxLat: 65.98468,
-      maxLon: -162.71869,
-      minLat: 60.97598,
-      minLon: -174.59527,
+      maxLat: 10,
+      maxLon: 100,
+      minLat: -10,
+      minLon: 90,
     };
-    expect(convertMapExtentToEnvelope(bounds)).toEqual({
-      "type": "envelope",
+    expect(convertMapExtentToPolygon(bounds)).toEqual({
+      "type": "polygon",
       "coordinates": [
-        [-174.59527, 65.98468], [-162.71869, 60.97598]
+        [[90, 10], [90, -10], [100, -10], [100, 10], [90, 10]]
       ]
     });
   });
 
   it('should clamp longitudes to -180 to 180', () => {
     const bounds = {
-      maxLat: 85.05113,
-      maxLon: 209.55801,
-      minLat: -85.05113,
-      minLon: -454.84711,
+      maxLat: 10,
+      maxLon: 200,
+      minLat: -10,
+      minLon: -400,
     };
-    expect(convertMapExtentToEnvelope(bounds)).toEqual({
-      "type": "envelope",
+    expect(convertMapExtentToPolygon(bounds)).toEqual({
+      "type": "polygon",
       "coordinates": [
-        [-180, 85.05113], [180, -85.05113]
+        [[-180, 10], [-180, -10], [180, -10], [180, 10], [-180, 10]]
       ]
     });
   });
 
-  it('should split bounds that cross dateline(east to west)', () => {
+  it('should handle bounds that cross dateline(east to west)', () => {
     const bounds = {
-      maxLat: 66.01959,
-      maxLon: 190.11434,
-      minLat: 61.0176,
-      minLon: 169.35168,
+      maxLat: 10,
+      maxLon: 190,
+      minLat: -10,
+      minLon: 170,
     };
-    expect(convertMapExtentToEnvelope(bounds)).toEqual([
-      {
-        "type": "envelope",
-        "coordinates": [
-          [169.35168, 66.01959], [180, 61.0176]
-        ]
-      },
-      {
-        "type": "envelope",
-        "coordinates": [
-          [-180, 66.01959], [-169.88566, 61.0176]
-        ]
-      },
-    ]);
+    expect(convertMapExtentToPolygon(bounds)).toEqual({
+      "type": "polygon",
+      "coordinates": [
+        [[170, 10], [170, -10], [-170, -10], [-170, 10], [170, 10]]
+      ]
+    });
   });
 
-  it('should split bounds that cross dateline(west to east)', () => {
+  it('should handle bounds that cross dateline(west to east)', () => {
     const bounds = {
-      maxLat: 14.29261,
-      maxLon: -159.0253,
-      minLat: -18.0925,
-      minLon: -193.69868,
+      maxLat: 10,
+      maxLon: -170,
+      minLat: -10,
+      minLon: -190,
     };
-    expect(convertMapExtentToEnvelope(bounds)).toEqual([
-      {
-        "type": "envelope",
-        "coordinates": [
-          [166.30132, 14.29261], [180, -18.0925]
-        ]
-      },
-      {
-        "type": "envelope",
-        "coordinates": [
-          [-180, 14.29261], [-159.0253, -18.0925]
-        ]
-      },
-    ]);
+    expect(convertMapExtentToPolygon(bounds)).toEqual({
+      "type": "polygon",
+      "coordinates": [
+        [[170, 10], [170, -10], [-170, -10], [-170, 10], [170, 10]]
+      ]
+    });
```

Comment:
consider adding a test for the case where the envelope is translated a couple rotations.

e.g. `[400,10,600,20]`

Reply:
mapbox will never give you values over past -360 and 360